### PR TITLE
Rename TagOrderStore

### PR DESCRIPTION
**Pull request by RinkiyaKeDad**
Hi! I've tried to fix issue #14670 by renaming TagPanel and TagOrderStore with GroupFilterPanel and GroupFilterOrderStore in the entire codebase. Please let me know if something needs to changed. Thanks! 

Fixes vector-im/element-web#14670

### CODE REVIEW

**Comment by t3chguy** [comment]
You have multiple merge conflicts, would be great to resolve those

Also as per the contributing doc, this needs a sign-off as per https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.rst#sign-off

File: CHANGELOG.md [hunk 1]
```diff
@@ -997,7 +997,7 @@ Changes in [2.7.0-rc.1](https://github.com/matrix-org/matrix-react-sdk/releases/
    [\#4347](https://github.com/matrix-org/matrix-react-sdk/pull/4347)
  * Revert "Prevent PersistedElements overflowing scrolled areas"
    [\#4682](https://github.com/matrix-org/matrix-react-sdk/pull/4682)
- * Remove unused TagPanelButtons
+ * Remove unused GroupFilterPanelButtons
```
Comment (turt2live) [inline CHANGELOG.md:1000]: please do not change mentions in the changelog

File: src/components/structures/CustomRoomGroupFilterPanel.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ import dis from '../../dispatcher/dispatcher';
 import classNames from 'classnames';
 import * as FormattingUtils from '../../utils/FormattingUtils';
 
-class CustomRoomTagPanel extends React.Component {
+class CustomRoomGroupFilterPanel extends React.Component {
```
Comment (turt2live) [inline src/components/structures/CustomRoomGroupFilterPanel.js:25]: The `CustomRoomTagPanel` was already correctly named - it should not be modified.

File: src/settings/Settings.ts [hunk 3]
```diff
@@ -331,11 +331,11 @@ export const SETTINGS: {[setting: string]: ISetting} = {
         displayName: _td('Mirror local video feed'),
         default: false,
     },
-    "TagPanel.enableTagPanel": {
+    "GroupFilterPanel.enableGroupFilterPanel": {
```
Comment (turt2live) [inline src/settings/Settings.ts:334]: this is a breaking change - we either need to be backwards compatible in the setting handlers or not rename the setting for now.

**Review by turt2live — CHANGES_REQUESTED** [review]

**Comment by RinkiyaKeDad** [comment]
Opened a [new PR](https://github.com/matrix-org/matrix-react-sdk/pull/5309) to fix this.

**Pull request closed without merging**